starlette_admin/fields.py

```diff
@@ -735,21 +737,39 @@ def input_params(self) -> str:
 
     async def parse_form_data(
         self, request: Request, form_data: FormData, action: RequestAction
-    ) -> Any:
+    ) -> Union[datetime, None]:
         try:
-            return datetime.fromisoformat(form_data.get(self.id))  # type: ignore
+            dt = datetime.fromisoformat(form_data.get(self.id))  # type: ignore
         except (TypeError, ValueError):
             return None
 
+        if dt.tzinfo is not None:
+            # Convert timezone-aware datetime to database timezone
+            database_tz = get_database_tzinfo()
+            return dt.astimezone(database_tz).replace(tzinfo=None)
+
+        # dt is naive, assume it's in the user's timezone
+        user_tz = get_tzinfo()
+        database_tz = get_database_tzinfo()
+        return dt.replace(tzinfo=user_tz).astimezone(database_tz).replace(tzinfo=None)
```

Comment:
This should only happen if the user explicitly sets the timezone config. It will make the transition easier.